### PR TITLE
[EC-1069] cannot save an item to a collection in the org vault view

### PR DESCRIPTION
**Pull request by coroiu**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fix collections not being ~loaded~ overridden properly in the org vault.

## Code changes

- **apps/web/src/app/organizations/vault/add-edit.component.ts**: Add method to allow overriding collections.
- **apps/web/src/app/organizations/vault/vault.component.ts**: Use new method to override collections.

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/app/organizations/vault/add-edit.component.ts [hunk 1]
```diff
@@ -68,6 +69,17 @@ export class AddEditComponent extends BaseAddEditComponent {
...
 
+  /**
+   * @deprecated Rewrite this entire modal to use `DialogService` and then use `DIALOG_DATA` instead.
+   * The dialog should probably also fetch its own collection instead of getting them from
+   * external components, so `this.collections` should be set to protected.
+   * For more information see: https://bitwarden.atlassian.net/browse/EC-1071
+   */
+  setCollections(collections: CollectionView[]) {
+    this.collections = collections;
+    this.writeableCollections = collections;
```
Comment (coroiu) [inline apps/web/src/app/organizations/vault/add-edit.component.ts:80]: questions: Does anybody know why we have both of these? The only place `this.collections` is still used seems to be in `loadCollections` where it allows external components to override the automatically loaded collections. But I can't really figure out why we don't just use `this.writeableCollections` for that.

```ts
  protected loadCollections() {
    if (!this.organization.canEditAnyCollection) {
      return super.loadCollections();
    }
    return Promise.resolve(this.collections);
  }```

File: apps/web/src/app/organizations/vault/vault.component.ts [hunk 2]
```diff
@@ -221,9 +221,11 @@ export class VaultComponent implements OnInit, OnDestroy {
...
     component.organizationId = this.organization.id;
     component.type = this.activeFilter.cipherType;
-    component.collections = (
-      await firstValueFrom(this.vaultFilterService.filteredCollections$)
-    ).filter((c) => !c.readOnly && c.id != null);
+    component.setCollections(
+      (await firstValueFrom(this.vaultFilterService.filteredCollections$)).filter(
+        (c) => !c.readOnly && c.id != null
+      )
+    );
```
Comment (coroiu) [inline apps/web/src/app/organizations/vault/vault.component.ts:228]: For whatever reason (most likely race conditions) the original code works fine in the personal vault, but not here in the org vault. I'm hesitant to fix things that aren't broken so I haven't touched the personal vault, but I'm thinking maybe I should anyways... Thoughts?

**Review by shane-melton — CHANGES_REQUESTED** [review]
I think you definitely found the culprit, but I think we might be able to solve the problem by removing some of the "inheritance spaghetti" which should reduce some of these unexpected side effects. It should also get us one step closer to an eventual solution using the new `DialogService`.

Please see my replies to your comments for more details.

**Comment by shane-melton** [comment]
Closing this in favor of https://github.com/bitwarden/clients/pull/4701

**Pull request closed without merging**